modules/library/main/src/main/java/org/geotools/filter/function/FilterFunction_equalTo.java

```diff
@@ -68,7 +82,21 @@ public Object evaluate(Object feature) {
             throw new IllegalArgumentException(
                     "Filter Function problem for function equalTo argument #1 - expected type Object");
         }
+        if (getParameters().size() > 2) {
+            try { // attempt to get value and perform conversion
+                matchAction = (MatchAction) getExpression(2).evaluate(feature, MatchAction.class);
+            } catch (Exception e) // probably a type error
+            {
+                throw new IllegalArgumentException(
+                        "Filter Function problem for function equalTo argument #2 - expected one of ANY, ONE or ALL");
+            }
+        }
+        Filter equalTo =
+                matchAction == null
+                        ? getFilterFactory2().equal(ff.literal(arg0), ff.literal(arg1), false)
+                        : getFilterFactory2()
+                                .equal(ff.literal(arg0), ff.literal(arg1), false, matchAction);
 
-        return Boolean.valueOf(StaticGeometry.equalTo(arg0, arg1));
+        return equalTo.evaluate(null);
```

Comment:
This is *so* much cleaner!

If you would like to be slightly more lazy there is no need to evaulate `arg0` and `arg1`, only to wrap them up as a literal again, if you replace this line with:
```
return equalTo.evaualte(feature)
```